File: sample-super-rx-ai/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/view/activity/UserDetailsActivity.java
package com.mateuszkoslacz.moviper.rxsample.viper.view.activity;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.mateuszkoslacz.moviper.base.view.activity.autoinject.passive.ViperLceAiPassiveActivity;
import com.mateuszkoslacz.moviper.iface.presenter.ViperPresenter;
import com.mateuszkoslacz.moviper.rxsample.R;
import com.mateuszkoslacz.moviper.rxsample.viper.contract.UserDetailsContract;
import com.mateuszkoslacz.moviper.rxsample.viper.entity.User;
import com.mateuszkoslacz.moviper.presentersdispatcher.MoviperPresentersDispatcher;

import butterknife.BindView;
import butterknife.ButterKnife;
import rx.subjects.PublishSubject;
import rx.subjects.Subject;

public class UserDetailsActivity
        // you can change base class to any Mosby Activity, ie. MvpLceActivity, MvpViewStateActivity, etc.
        extends ViperLceAiPassiveActivity
        <LinearLayout,
                User,
                UserDetailsContract.View>
        implements UserDetailsContract.View, UserDetailsContract.ViewHelper {

    public final static String USER_EXTRA = "USER_EXTRA";

    @BindView(R.id.login)
    TextView mLoginTextView;
    @BindView(R.id.avatar)
    ImageView mAvatarImageView;
    @BindView(R.id.url)
    TextView mUrlTextView;
    @BindView(R.id.name)
    TextView mNameTextView;
    @BindView(R.id.company)
    TextView mCompanyTextView;
    @BindView(R.id.blog)
    TextView mBlogTextView;
    @BindView(R.id.location)
    TextView mLocationTextView;
    @BindView(R.id.email)
    TextView mEmailTextView;

    PublishSubject<String> mAvatarClicks = PublishSubject.create();

    public static void start(Context context, User user) {
        context.startActivity(getStartingIntent(context, user));
    }

    public static Intent getStartingIntent(Context context, User user) {
        Intent starter = new Intent(context, UserDetailsActivity.class);
        starter.putExtra(USER_EXTRA, user.getLogin());
        return starter;
    }

    @Override
    public void bindDataToViews(User user) {
        mLoginTextView.setText(user.getLogin());
        mUrlTextView.setText(user.getUrl());
        mNameTextView.setText(user.getName());
        mCompanyTextView.setText(user.getCompany());
        mBlogTextView.setText(user.getBlog());
        mLocationTextView.setText(user.getLocation());
        mEmailTextView.setText(user.getEmail());
        Glide.with(this)
                .load(user.getAvatarUrl())
                .into(mAvatarImageView);
        mAvatarImageView.setOnClickListener(v -> mAvatarClicks.onNext(user.getAvatarUrl()));
    }

    @Override
    public Subject<String, String> getAvatarClicks() {
        return mAvatarClicks;
    }

    @Override
    public View getAvatarImageView() {
        return mAvatarImageView;
    }

    @Override
    public void setData(User user) {
        bindDataToViews(user);
    }

    @Override
    protected String getErrorMessage(Throwable e, boolean pullToRefresh) {
        return e.getLocalizedMessage();
    }

    @Override
    public void loadData(boolean pullToRefresh) {
    }

    @NonNull
    @Override
    public ViperPresenter<UserDetailsContract.View> createPresenter() {
        return MoviperPresentersDispatcher.getInstance().getPresenterForView(this);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_user_details;
    }

    @Override
    protected void injectViews() {
        super.injectViews();
        ButterKnife.bind(this);
    }
}

File: sample-super-rx-ai-kotlin/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/view/activity/UserDetailsActivity.kt
package com.mateuszkoslacz.moviper.rxsample.viper.view.activity

import android.content.Context
import android.content.Intent
import android.view.View
import android.widget.LinearLayout
import com.bumptech.glide.Glide
import com.mateuszkoslacz.moviper.base.view.activity.autoinject.passive.ViperLceAiPassiveActivity
import com.mateuszkoslacz.moviper.iface.presenter.ViperPresenter
import com.mateuszkoslacz.moviper.rxsample.R
import com.mateuszkoslacz.moviper.rxsample.viper.contract.UserDetailsContract
import com.mateuszkoslacz.moviper.rxsample.viper.entity.User
import com.mateuszkoslacz.moviper.server.MoviperPresentersDispatcher
import kotlinx.android.synthetic.main.activity_user_details.*
import rx.Observable
import rx.subjects.PublishSubject

class UserDetailsActivity :
        ViperLceAiPassiveActivity<LinearLayout, User, UserDetailsContract.View>(),
        UserDetailsContract.View, UserDetailsContract.ViewHelper {

    override val avatarClicks: Observable<String>
        get() = mAvatarClicks

    override val avatarImageView: View
        get() = avatar

    internal var mAvatarClicks = PublishSubject.create<String>()

    override fun bindDataToViews(user: User) {
        login?.text = user.login
        url?.text = user.url
        name?.text = user.name
        company?.text = user.company
        blog?.text = user.blog
        location?.text = user.location
        email?.text = user.email
        Glide.with(this)
                .load(user.avatarUrl)
                .into(avatar)
        avatar?.setOnClickListener { mAvatarClicks.onNext(user.avatarUrl) }
    }

    override fun setData(user: User) = bindDataToViews(user)

    override fun getErrorMessage(e: Throwable, pullToRefresh: Boolean): String = e.message!!

    override fun loadData(pullToRefresh: Boolean) {}

    override fun createPresenter(): ViperPresenter<UserDetailsContract.View> =
            MoviperPresentersDispatcher.getInstance().getPresenterForView(this)
                    as ViperPresenter<UserDetailsContract.View>

    override fun getLayoutId(): Int = R.layout.activity_user_details

    companion object {

        val USER_EXTRA = "USER_EXTRA"

        fun start(context: Context, user: User) {
            context.startActivity(getStartingIntent(context, user))
        }

        fun getStartingIntent(context: Context, user: User): Intent {
            val starter = Intent(context, UserDetailsActivity::class.java)
            starter.putExtra(USER_EXTRA, user.login)
            return starter
        }
    }
}
